code/controllers/subsystem/ships.dm

```diff
@@ -90,6 +90,8 @@ var/datum/subsystem/ship/SSship
 		if(S.shield_strength >= initial(S.shield_strength))
 			if(S.attacking_player && S.shield_strength > starting_shields) broadcast_message("<span class=notice>Enemy ship ([S.name]) has recharged shields to 100% strength.</span>",notice_sound)
 
+	if(!find_broken_components(S))
+		S.next_repair = world.time + S.repair_time
```

Comment:
It's not cumulative. It adds repair time to world time. Repair time is always the same.